test_conformance/extensions/cl_khr_command_buffer/command_buffer_test_event_info.cpp

```diff
@@ -0,0 +1,240 @@
+    size_t size;
+};
+
+struct Context : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, &event);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        error = clWaitForEvents(1, &event);
+        test_error(error, "clWaitForEvents failed");
+
+        error = clGetEventInfo(event, CL_EVENT_CONTEXT, sizeof(ret_context),
+                               &ret_context, &size);
+        test_error(error, "clGetEventInfo failed");
+
+        if (ret_context != context)
+        {
+            log_error("ERROR: Wrong context returned by clGetEventInfo");
+            return TEST_FAIL;
+        }
+
+        return CL_SUCCESS;
+    }
+
+    clEventWrapper event;
+    cl_context ret_context = nullptr;
+    size_t size;
+};
+
+struct ExecutionStatus : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, &event);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        error = clWaitForEvents(1, &event);
+        test_error(error, "clWaitForEvents failed");
+
+        error = clGetEventInfo(event, CL_EVENT_COMMAND_EXECUTION_STATUS,
+                               sizeof(status), &status, NULL);
+        test_error(error, "clGetEventInfo failed");
+
+        if (status != CL_COMPLETE)
+        {
+            log_error(
+                "ERROR: Incorrect status returned from clGetEventInfo (%d)\n",
+                status);
+            return TEST_FAIL;
+        }
+
+        return CL_SUCCESS;
+    }
+
+    clEventWrapper event;
+    cl_int status;
+};
+
+struct ReferenceCount : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, &event);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        error = clWaitForEvents(1, &event);
+        test_error(error, "clWaitForEvents failed");
+
+        error = clGetEventInfo(event, CL_EVENT_REFERENCE_COUNT, sizeof(count),
+                               &count, &size);
+        test_error(error, "clGetEventInfo failed");
+
+        if (count != expected_count)
```

Comment:
The wording around spec reference counts means that I think it would still be legal for a implementation to return a value larger than 1 here, I'd just make this check `count > 0`